components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.management.apis;
+
+import org.apache.axis2.Constants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.json.JSONObject;
+import org.wso2.micro.core.util.StringUtils;
+import org.wso2.micro.integrator.initializer.handler.DataHolder;
+import org.wso2.micro.integrator.initializer.handler.transaction.store.TransactionStore;
+import org.wso2.micro.integrator.initializer.handler.transaction.store.connector.RDBMSConnector;
+
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.wso2.micro.integrator.management.apis.Constants.BAD_REQUEST;
+import static org.wso2.micro.integrator.management.apis.Constants.INTERNAL_SERVER_ERROR;
+
+/**
+ * Resource for a retrieving aggregated request count.
+ * <p>
+ * Handles resources in the form "management/reqCount and management/reqCount?year=2020&month=5"
+ */
+public class RequestCountResource extends ApiResource {
+
+    private static Log log = LogFactory.getLog(RequestCountResource.class);
+    private static final String NO_ENTITY_BODY = "NO_ENTITY_BODY";
+
+    /**
+     * Constructor for creating an API Resource.
+     *
+     * @param urlTemplate the url template of the Resource
+     */
+    public RequestCountResource(String urlTemplate) {
+
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+
+        Set<String> methods = new HashSet<>();
+        methods.add("GET");
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext synCtx) {
+
+        org.apache.axis2.context.MessageContext axisCtx = ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+        String yearParameter = Utils.getQueryParameter(synCtx, "year");
+        String monthParameter = Utils.getQueryParameter(synCtx, "month");
+
+        if (tryParseInt(yearParameter) != null && tryParseInt(monthParameter) != null) {
+            if (!StringUtils.isEmpty(yearParameter) && !StringUtils.isEmpty(monthParameter)) {
```

Comment:
can it be empty at this point? Wouldn't a numberformatException be thrown inside the method tryParseInt and a null be returned?

Reply:
I wrote a method called tryParseInt to return null in the case of numberformatException

Reply:
Exactly. So, can yearParameter/monthParameter be empty without tryParseInt(yearParameter) returning null?